lib/membrane_rtc_engine/endpoints/webrtc/variant_selector.ex

```diff
@@ -174,8 +300,78 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.VariantSelector do
         :high -> 3
         :medium -> 2
         :low -> 1
+        :no_variant -> 0
       end,
       :desc
     )
   end
+
+  defp perform_automatic_layer_selection(%__MODULE__{} = selector) do
+    selector
+    |> select_variant()
+    |> tap(&manage_allocation/1)
+  end
+
+  defp manage_allocation({selector, _variant}), do: manage_allocation(selector)
+
+  defp manage_allocation(%__MODULE__{} = selector) do
+    current_variant_bitrate = selector.variant_bitrates[selector.current_variant]
+    queued_variant_bitrate = selector.variant_bitrates[selector.queued_variant] || 0
+
+    required_bitrate =
+      [current_variant_bitrate, queued_variant_bitrate]
+      |> Enum.max()
+
+    next_variant = next_desired_variant(selector)
+
+    cond do
+      # If we're not having that much margin left in current allocation,
+      # try to request higher allocation
+      # It is very important that this clause takes precedence
+      # over the one that aims to increase quality.
+      # If we don't have enough bandwidth to maintain current quality,
+      # don't bother with better quality, try to salvage current quality
+      required_bitrate > 0.95 * selector.current_allocation ->
+        Membrane.Logger.debug(
+          "Requesting #{required_bitrate / 1024} kbps from connection prober as a mean to maintain current quality"
```

Comment:
```suggestion
          "Requesting #{required_bitrate * 1.1 / 1024} kbps from connection prober as a mean to maintain current quality"
```
Please take a look at other logs too